src/Apps/Artwork/Components/ArtworkBanner/index.tsx

```diff
@@ -0,0 +1,152 @@
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import { ContextConsumer } from "Artsy/Router"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { get } from "Utils/get"
+import { Banner } from "./Banner"
+
+import { ArtworkBanner_artwork } from "__generated__/ArtworkBanner_artwork.graphql"
+import { ArtworkBannerQuery } from "__generated__/ArtworkBannerQuery.graphql"
+
+export interface ArtworkBannerProps {
+  artwork: ArtworkBanner_artwork
+}
+
+export const ArtworkBanner: React.SFC<ArtworkBannerProps> = props => {
+  const { context, partner } = props.artwork
+  if (!context) return null
+  // imageUrl: image for avatar
+  // initials: fallback partner initials in case image is not there.
+  // meta: in auction / at fair / in show
+  // name:  auction / fair / show name
+  // subHeadline: partner name
+  switch (context.__typename) {
+    case "ArtworkContextAuction":
+      const auctionImage = get(partner, p => p.profile.icon.url)
+      return (
+        <Banner
+          imageUrl={auctionImage}
+          initials={partner.initials}
+          meta="In auction"
+          name={context.name}
+          subHeadline={partner.name}
+        />
+      )
+    case "ArtworkContextFair":
+      const fairImage = get(context, c => c.profile.icon.img.url)
+      const initials = get(context, c => c.profile.initials)
+      return (
+        <Banner
+          imageUrl={fairImage}
+          initials={initials}
+          meta="At fair"
+          name="Fair name goes here" // {context.name}
```

Comment:
@alloy - so name would display properly here, but will give me an error and would not allow to push:)